site/docs/relations/logical_relations.md

```diff
@@ -57,6 +57,16 @@ is encoded as records consisting of literal values.
 | -------- | ----------- | -------- |
 | Data     | Required    | Required |
 
+#### Virtual Expression Table
+
+A virtual expression table is a table whose contents are embedded in the plan itself.  The table data
+is encoded as records consisting of expression values.
+
+| Property | Description | Required |
+| -------- | ----------- | -------- |
+| Data     | Required    | Required |
```

Comment:
I think an important property of a virtual (expression) table is that it doesn't contain any field references.  This may not be obvious to new readers to Substrait.  In other words, they may look at a virtual table and think "this is weird, it looks like a projection?  Maybe this is some kind of select relation?"

Maybe something like...

```
A virtual expression table is a table whose contents are embedded in the plan itself.  Each column is an expression that can be resolved without referencing any input data.  These expressions will not contain field references but may contain function calls, literals, and other expression nodes.
```

Reply:
Adding to this, I assume that these expression shouldn't contain other relations (i.e. subqueries)?

Reply:
I don't think we should have prohibitions against either (field references and subqueries).

That being said, a valid field reference in this context is impossible because there are zero fields to reference. So having field[47] here is just as illegal as having a readrel that has two columns and trying to do field[47].

Reply:
As a fun aside, in Snowflake the following is legal.
```
select * from (values ((select 1 where (select true)), 'Hello'||'World'))
```

Reply:
The problem with nested queries having a field reference is that the resulting structure is no longer scalar.  For systems without virtual table support I have had to write tables to disk and then register named tables.  For those systems I would have to reject field references.

Reply:
> I don't think we should have prohibitions against either (field references and subqueries).

That's fine.  I don't know that we need to go so far as prohibit things.  My goal is just to make it as obvious as possible what the message is used for.  I'd be find with wording like...

```
A virtual expression table is a table whose contents are embedded in the plan itself.
Each column is an expression that can be resolved without referencing any input data.
For example, a literal, a function call involving literals, or any other expression that does
not require input.
```

Reply:
> I'd be find with wording like
Fixed doc to indicate virtual expression table are not dependent on any input